iota/api.py

```diff
@@ -749,15 +751,293 @@ def were_addresses_spent_from(self, addresses):
 
         - https://docs.iota.org/docs/node-software/0.1/iri/references/api-reference#wereaddressesspentfrom
         """
-        return core.WereAddressesSpentFromCommand(self.adapter)(
+        return await core.WereAddressesSpentFromCommand(self.adapter)(
             addresses=addresses,
         )
 
+class StrictIota(AsyncStrictIota):
+    """
+    Synchronous API to send HTTP requests for communicating with an IOTA node.
+
+    This implementation only exposes the "core" API methods.  For a more
+    feature-complete implementation, use :py:class:`Iota` instead.
+
+    References:
+
+    - https://docs.iota.org/docs/node-software/0.1/iri/references/api-reference
+
+    :param AdapterSpec adapter:
+        URI string or BaseAdapter instance.
+
+    :param Optional[bool] devnet:
+        Whether to use devnet settings for this instance.
+        On the devnet, minimum weight magnitude is set to 9, on mainnet
+        it is 1 by default.
+
+    :param Optional[bool] local_pow:
+        Whether to perform proof-of-work locally by redirecting all calls
+        to :py:meth:`attach_to_tangle` to
+        `ccurl pow interface <https://pypi.org/project/PyOTA-PoW/>`_.
+
+            See :ref:`Optional Local Pow` for more info and
+            :ref:`find out<pow-label>` how to use it.
+
+    """
+
+    def __init__(self, adapter, devnet=False, local_pow=False):
+        # type: (AdapterSpec, bool, bool) -> None
+        # Copy parent's doctrsing
+        __doc__ = super(StrictIota, self).__init__.__doc__
+        super(StrictIota, self).__init__(adapter, devnet, local_pow)
+
+
+    def add_neighbors(self, uris):
+        # type: (Iterable[Text]) -> dict
+        # Copy parent's docstring
+        __doc__ = super(StrictIota, self).add_neighbors.__doc__
+
+        # Execute original coroutine inside an event loop to make this method
+        # synchronous
+        return asyncio.get_event_loop().run_until_complete(
+            super(StrictIota, self).add_neighbors(uris)
```

Comment:
agreed :+1:

Reply:
btw, @todofixthis could you explain me why `super()` is called in a base class?
For example here:
```python
@add_metaclass(AdapterMeta)
class BaseAdapter(object):
    """
    Interface for IOTA API adapters.

    Adapters make it easy to customize the way an API instance
    communicates with a node.
    """
    supported_protocols = ()  # type: Tuple[Text]
    """
    Protocols that ``resolve_adapter`` can use to identify this adapter
    type.
    """

    def __init__(self):
        super(BaseAdapter, self).__init__()
```

Reply:
🤔 Good question.  I think the reason is in case a derived class uses multiple inheritance, e.g., `CustomAdapter(BaseAdapter, OtherBase):` (or possibly the other way around; multiple inheritance is so confusing 😅) — without the super call, it's possible that the constructor for one or more bases won't get called.

Reply:
Okay, thanks for the answer!

Let's say we switch to py3-style `super()` calls everywhere and don't need to explicitly tell `type` and `object` to `super()`. If I do it with `BaseAdapter`:
```python
@add_metaclass(AdapterMeta)
class BaseAdapter:
    """
    Interface for IOTA API adapters.

    Adapters make it easy to customize the way an API instance
    communicates with a node.
    """
    supported_protocols = ()  # type: Tuple[Text]
    """
    Protocols that ``resolve_adapter`` can use to identify this adapter
    type.
    """

    def __init__(self):
        super().__init__()
``` 
I get a `TypeError` in the `super()` call:
```
Traceback (most recent call last):
  File "/pyota/test/commands/extended/utils_test.py", line 16, in setUp
    self.adapter = MockAdapter()
  File "/pyota/iota/adapter/__init__.py", line 547, in __init__
    super(MockAdapter, self).__init__()
  File "/pyota/iota/adapter/__init__.py", line 172, in __init__
    super().__init__()
TypeError: super(type, obj): obj must be an instance or subtype of type
```
, but the MRO seems okay:
```
>>> from iota import BaseAdapter
>>> BaseAdapter.__mro__
(<class 'iota.adapter.BaseAdapter'>, <class 'object'>)
```
What am I missing here? :thinking:

Reply:
Thanks for checking that out!  Hmmm.. maybe it's not needed in Python 3 then?  I don't recall the exact situation where I thought it was necessary; maybe we just remove it and see if anyone complains 😅